### PR TITLE
HMW-664 Switched from react-window to react-virtuoso

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-664](https://jira.epa.gov/browse/HMW-659)

## Main Changes:
* Switched from using the `react-window` package to [`react-virtuoso`](https://github.com/petyosi/react-virtuoso) for smoother list virtualization. The `react-virtuoso` package is smaller, it uses `ResizeObserver` out of the box (for dynamic content like the CyAN content), and otherwise requires no imperative item sizing.

## Steps To Test:
1. Go to http://192.168.0.244:3000/community/lake%20mattamuskeet/overview, and click on _Water Monitoring Locations_.
2. With all monitoring types selected, scroll down to the CyAN waterbody. The list should scroll smoothly along the way.
3. Expand the CyAN content, and wait for the content to load. The accordion item should resize to fit the content.
4. Test the behavior of the other virtualized list instances, i.e. the characteristics filter, the waterbody list, and the list of waters on the Plan Summary page.

### CODE REVIEW

File: app/client/src/components/shared/MenuList.tsx [hunk 1]
```diff
@@ -36,44 +22,13 @@ export function MenuList<T>(props: MenuListProps<T>) {
...
-              index={index}
-              width={width}
-              setSize={setSize}
-              value={childrenToDisplay[index]}
-            />
-          </div>
-        )}
-      </VariableSizeList>
+      <Virtuoso
+        itemContent={(index) => <div>{childrenToDisplay[index]}</div>}
```
Comment (cschwinderg) [inline app/client/src/components/shared/MenuList.tsx:26]: I'm seeing some performance issues with this. 

The `Waterbody Names` menu is very slow to scroll on the http://localhost:3000/state/PA/advanced-search page. There are 216,159 waterbodies in that list.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
I like how much simpler `react-virtuoso` is! Works well on most pages. 

I am seeing some performance issues on the state advanced search page for PA.

File: app/client/src/components/shared/MenuList.tsx (hunk 1 above)
Comment (maxdiebold-erg) [reply app/client/src/components/shared/MenuList.tsx:26]: Good call, the menu isn't even opening for me, I will look into this.

**Comment by maxdiebold-erg** [comment]
I thought the sluggishness was due to `react-virtuoso`, but it looks like Pennsylvania has the same issue in production, so it seemed to be a problem with `react-select`. [Multiple](https://github.com/JedWatson/react-select/issues/3128) [issues](https://github.com/JedWatson/react-select/issues/3385) point to the `onMouseDown` and `onMouseMove` handlers as the issues, so I tried to incorporate some ideas in those threads.

With the changes, the scrolling is a lot faster but opening the menu is still a bit slow. Also, the fixes break any styles that relied on `isFocused` or `isSelected` state: this doesn't affect any of our uses, since the virtualized `MenuList` is only used for multi-selects (where `isSelected` items are not styled) and the keyboard selection is was already broken in virtualized select menus (the menu would not follow the cursor past the first visible items). There are other props to help with this in `react-select` custom components, so I'll address the focus styling once keyboard functionality is fixed.

**Comment by maxdiebold-erg** [comment]
It looks like `react-select` doesn't scroll to follow focus with the arrow keys to begin with. I think we can address this, but may want to do so in a different ticket. For now, I will just work on restoring the focus and "selected" styling.

**Comment by maxdiebold-erg** [comment]
As an alternative to trying to hack `react-select`, we could also load options asynchronously like we do in Expert Query. In this case, there wouldn't be any network calls, it would just reference the options stored elsewhere in a variable. That would take care of the slow initial loading of the select input, too.

**Comment by cschwinderg** [comment]
For me it was slow on production as well, but not nearly as bad. I couldn't scroll at all on the local version. 

I like your idea of doing it asynchronously like EQ. If we go that route, would we even need `react-virtuoso` for the `react-select`?

**Comment by maxdiebold-erg** [comment]
For EQ, we still use a virtualized list because the number of options in the menu increases as the user scrolls. But I think it should be straightforward enough to just spit back the current page and not add the page to the previous results, in which case we could get rid of the virtualization.

**Comment by maxdiebold-erg** [comment]
I'll finish this up early Monday. Almost there but need to combine the chunking with the search filter.

**Comment by maxdiebold-erg** [comment]
@cschwinderg This should be good to review again. I ended up sticking with virtualization in the select menus because `react-select-async-paginate` really doesn't seem suited to paginating in both directions, and this way avoids another dependency.

**Review by cschwinderg — APPROVED** [review]
Performance is much better now!

**Pull request merged**